src/http-manager.ts

```diff
@@ -38,4 +38,22 @@ export default class HttpManager {
   getLatestResponseId(responseName: ServiceType): number {
     return this.latestResponseIds[responseName];
   }
+
+  /**
+   * Update the recorded received response id of the given service type if
+   * the given request id corresponds to the latest response.
```

Comment:
what do you think of rewording to 'the given request id is newer than the latest saved response id?'